tests/unit/test_sharpspring.py

```diff
@@ -0,0 +1,102 @@
+from unittest.mock import patch
+
+from pmg.sharpspring import Sharpspring
+from tests import PMGTestCase
+from tests.fixtures import dbfixture, UserData, OrganisationData
+
+
+class MockResponse:
+    def __init__(self, json_data, status_code):
+        self.json_data = json_data
+        self.status_code = status_code
+
+    def raise_for_status(self):
+        pass
+
+    def json(self):
+        return self.json_data
+
+
+def get_requests_post_success(*args, **kwargs):
+    return MockResponse(
+        {"result": {"creates": [{"success": True,}]}, "error": False}, 200
+    )
+
+
+def get_sharpspring_call_success(*args, **kwargs):
+    return {"result": {"creates": [{"success": True,}]}, "error": False}
+
+
+def get_sharpspring_call_fail_create_lead(*args, **kwargs):
+    return {"result": {"creates": [{"success": False, "error": {"code": 500}}]}}
```

Comment:
does this not raise in raise_for_status?

Reply:
My mock response doesn't: https://github.com/OpenUpSA/pmg-cms-2/pull/401/files#diff-90beb5472c712ae3ffaa178e243ca4c9R13-R14

Since the sharpspring code checks for error codes I'm assuming there must be instances where `raise_for_status` doesn't raise an error, but there is still some error code. It probably won't be a 500 code, but I haven't looked at the Sharpspring docs or anything to see what it might be. 

https://github.com/OpenUpSA/pmg-cms-2/pull/401/files#diff-ca4d79d8ac81dcd09cb9a1137ab1fcfcR47

Reply:
Looks like "Object-level" errors won't raise an error according to the [Sharpspring docs](https://help.sharpspring.com/hc/en-us/articles/115001069228-Understanding-SharpSpring-Open-API-Overview#h_8587766891311518636374378).